internal/configs/bootnode/bootnode.go

```diff
@@ -0,0 +1,316 @@
+	if err != nil {
+		idleTimeout, _ = time.ParseDuration(nodeconfig.DefaultHTTPTimeoutIdle)
+		utils.Logger().Warn().
+			Str("provided", bnc.HTTP.IdleTimeout).
+			Dur("updated", idleTimeout).
+			Msg("Sanitizing invalid http idle timeout")
+	}
+	evmCallTimeout, err := time.ParseDuration(bnc.RPCOpt.EvmCallTimeout)
+	if err != nil {
+		evmCallTimeout, _ = time.ParseDuration(nodeconfig.DefaultEvmCallTimeout)
+		utils.Logger().Warn().
+			Str("provided", bnc.RPCOpt.EvmCallTimeout).
+			Dur("updated", evmCallTimeout).
+			Msg("Sanitizing invalid evm_call timeout")
+	}
+	return nodeconfig.RPCServerConfig{
+		HTTPEnabled:        bnc.HTTP.Enabled,
+		HTTPIp:             bnc.HTTP.IP,
+		HTTPPort:           bnc.HTTP.Port,
+		HTTPAuthPort:       bnc.HTTP.AuthPort,
+		HTTPTimeoutRead:    readTimeout,
+		HTTPTimeoutWrite:   writeTimeout,
+		HTTPTimeoutIdle:    idleTimeout,
+		WSEnabled:          bnc.WS.Enabled,
+		WSIp:               bnc.WS.IP,
+		WSPort:             bnc.WS.Port,
+		WSAuthPort:         bnc.WS.AuthPort,
+		DebugEnabled:       bnc.RPCOpt.DebugEnabled,
+		PreimagesEnabled:   bnc.RPCOpt.PreimagesEnabled,
+		EthRPCsEnabled:     bnc.RPCOpt.EthRPCsEnabled,
+		StakingRPCsEnabled: bnc.RPCOpt.StakingRPCsEnabled,
+		LegacyRPCsEnabled:  bnc.RPCOpt.LegacyRPCsEnabled,
+		RpcFilterFile:      bnc.RPCOpt.RpcFilterFile,
+		RateLimiterEnabled: bnc.RPCOpt.RateLimterEnabled,
+		RequestsPerSecond:  bnc.RPCOpt.RequestsPerSecond,
+		EvmCallTimeout:     evmCallTimeout,
+	}
+}
+
+type DnsSync struct {
+	Port       int    // replaces: Network.DNSSyncPort
+	Zone       string // replaces: Network.DNSZone
+	Client     bool   // replaces: Sync.LegacyClient
+	Server     bool   // replaces: Sync.LegacyServer
+	ServerPort int
+}
+
+type NetworkConfig struct {
+	NetworkType string
+	BootNodes   []string
+}
+
+type P2pConfig struct {
+	Port                 int
+	IP                   string
+	KeyFile              string
+	DHTDataStore         *string `toml:",omitempty"`
+	DiscConcurrency      int     // Discovery Concurrency value
+	MaxConnsPerIP        int
+	DisablePrivateIPScan bool
+	MaxPeers             int64
+	// In order to disable Connection Manager, it only needs to
+	// set both the high and low watermarks to zero. In this way,
+	// using Connection Manager will be an optional feature.
+	ConnManagerLowWatermark  int
+	ConnManagerHighWatermark int
+	WaitForEachPeerToConnect bool
+	// to disable p2p security (tls and noise)
+	NoTransportSecurity bool
+	// enable p2p NAT. NAT Manager takes care of setting NAT port mappings, and discovering external addresses
+	NAT bool
+	// custom user agent; explicitly set the user-agent, so we can differentiate from other Go libp2p users
+	UserAgent string
+	// p2p dial timeout
+	DialTimeout time.Duration
+	// P2P multiplexer type, should be comma separated (mplex, Yamux)
+	Muxer string
+	// No relay services, direct connections between peers only
+	NoRelay bool
+}
+
+type GeneralConfig struct {
```

Comment:
i think only NodeType and ShardID is really required here

Reply:
removed